### ISSUE TITLE
use x-hub-signature to validate webhook JSON integrity

### ISSUE THREAD

**Issue by danielhartnell** [mozilla-iam/aws-github-to-s3#14]
Amazon's git2s3 quick start did a wonderful job addressing this:

https://github.com/aws-quickstart/quickstart-git2s3/blob/master/functions/source/GitPullS3/lambda_function.py#L145-L150

The same type of validation will need to be added to `handler.py`. As a general note, I should also write a helper method to sanitize input and clean anything that comes in from that webhook to avoid potential security issues. Couldn't hurt to find and create a regex that aligns with allowed characters for GitHub user / org and repository names.

To summarize, here are some general security improvements:

- [ ] Sanitize any input pulled from the JSON payload
- [ ] Validate integrity of payload
